homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,179 @@
+        name = device["name"]
+        openlevel = device["openlevel"]
+        movement = device["movement"]
+        connected = device["connected"]
+        model = device["model"]
+        device_class = CoverDeviceClass.SHUTTER
+
+        _LOGGER.info(
+            "Adding DIO Chacon SHUTTER Cover with id=%s, name=%s, openlevel=%s, movement=%s and connected=%s",
+            target_id,
+            name,
+            openlevel,
+            movement,
+            connected,
+        )
+
+        super().__init__(coordinator, target_id, name, model, connected)
+
+        self._attr_current_cover_position = openlevel
+        self._attr_is_closing = movement == ShutterMoveEnum.DOWN.value
+        self._attr_is_opening = movement == ShutterMoveEnum.UP.value
+        self._attr_device_class = device_class
+
+        self._attr_supported_features = (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.STOP
+            | CoverEntityFeature.SET_POSITION
+        )
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return The openlevel of the cover un percentage."""
+        if self.coordinator_data and self.coordinator_data["openlevel"]:
+            self._attr_current_cover_position = self.coordinator_data["openlevel"]
+        return self._attr_current_cover_position
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed."""
+        return self.current_cover_position == 0
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing."""
+        if self.coordinator_data and self.coordinator_data["movement"]:
+            self._attr_is_closing = (
+                self.coordinator_data["movement"] == ShutterMoveEnum.DOWN.value
+            )
+        return self._attr_is_closing
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening."""
+        if self.coordinator_data and self.coordinator_data["movement"]:
+            self._attr_is_opening = (
+                self.coordinator_data["movement"] == ShutterMoveEnum.UP.value
+            )
+        return self._attr_is_opening
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+
+        _LOGGER.debug(
+            "Close cover %s , %s, %s",
+            self._target_id,
+            self._attr_name,
+            self.is_closed,
+        )
+
+        # closes effectively only if cover is not already closing and not fully closed
+        if not self._attr_is_closing and not self.is_closed:
+            self._attr_is_closing = True
+            self.async_write_ha_state()
+
+            await self.dio_chacon_client.move_shutter_direction(
+                self._target_id, ShutterMoveEnum.DOWN
+            )
+
+        # Closed signal is managed via the callback that triggers async_set_updated_data on the coordinator
```

Comment:
What does this comment refer to

Reply:
Change it.
It means that the final cover status that will be "closed" is not set here at the end of the function but after the callback call that will send a cover stop position.